{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "military-binary",
   "metadata": {},
   "source": [
    "# Overview\n",
    "\n",
    "Preprocessing of ASAP-seq and DOGMA-seq datasets.\n",
    "\n",
    "## Documentation on preprocessing from Mimitou publication (methods) and online code\n",
    "For reference.\n",
    "### ASAP-seq\n",
    "> \"Similarly, we identified high-quality cells from the ASAP-seq dataset such that each cell had a TSS score exceeding 4 and a minimum of 1,000 fragments. Cells were further filtered out if they had excess abundance of total protein tags (>25,000 in either condition) or tags measured from the rat isotype controls (>75 in either condition)\" From Methods in Mimitou et al., 2021\n",
    "\n",
    "#### ATAC\n",
    "- TSS enrichment > 4 (methods & [online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/02_ArchR_firstpass.R))\n",
    "- fragments > 1000 (methods & [online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/02_ArchR_firstpass.R))\n",
    "- doublet removal with ArchR ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/02_ArchR_firstpass.R))\n",
    "\n",
    "#### ADT \n",
    "- counts >= 100 ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/01_ADT_preprocess_ASAP.R))\n",
    "- counts <= 25000 (methods & [online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/01_ADT_preprocess_ASAP.R))\n",
    "- counts for rat isotype control <= 75 (methods & [online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stimulation_asapseq_citeseq/code/01_ADT_preprocess_ASAP.R))\n",
    "\n",
    "\n",
    "### DOGMA-seq\n",
    "#### GEX\n",
    "- pct_mito < 30 ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "- n_total_rna > 1000 ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "- n_feature_rna > 500 ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "- nCount_RNA < (10^4.5)\n",
    "\n",
    "#### ADT\n",
    "- totalCTRLadt < 10 (https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "- totalADT > 100 (https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "- !(full_meta$CD8adt > 30 & full_meta$CD4adt > 100)) (mutually exclusive) (https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R))\n",
    "\n",
    "#### ATAC\n",
    "- pct_in_peaks > 50 ([online code](https://github.com/caleblareau/asap_reproducibility/blob/master/pbmc_stim_multiome/code/11_setup.R)) - this seems to rely on CellRanger preprocessing/output for peaks, etc. for the ATAC data, ArchR was not used\n",
    "\n",
    "\n",
    "\n",
    "## Own preprocessing choices\n",
    "Attempt to harmonize preprocessing, if anything, be more strict!:\n",
    "- no doublet filtering for ASAP-seq ATAC data\n",
    "- data adaptive upper thresholds for ATAC and GEX data\n",
    "- Unified informative feature finding strategy for ATAC (Mimitou: ArchR for ASAP-seq and CellRanger for DOGMA-seq) - I am using orthogonal method scregseg\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "analyzed-salmon",
   "metadata": {},
   "source": [
    "For simplicity, I prefilter all ATAC data based on what was done for ASAP seq, then call regions with ScregSeg\n",
    "\n",
    "Afterward, I will remove cells from that if paired ADT or RNA is not well, however, for region calling the data should be good enough.\n",
    "\n",
    "\n",
    "\n",
    "As I want to integrate both of these different datasets, I want to apply as similar preprocessing as possible.\n",
    "\n",
    "I dedcided to be as stringent as possible and apply all filters of both (where possible)\n",
    "\n",
    "I will not filter doublets with ArchR (as was done for ASAP-seq), but only make use of the \"soft\" doublet removal through making high cd4 cd8 counts mutually exclusive.\n",
    "\n",
    "### ASAP-seq and DOGMA-seq (applicable modalities)\n",
    "#### ATAC \n",
    "- TSS score exceeding 4 \n",
    "- minimum of 1,000 fragments\n",
    "- Additionally, after this initial filtering remove high counts based on Q3 + 1.5x IQR (after having removed low counts!)\n",
    "\n",
    "All these cells are used for finding informative regions with scregseg (detailed below), however, these cells might be dropped later for not meeting the threshold of the other modalities!\n",
    "\n",
    "#### ADT\n",
    " - total tags need to be <= 25,000 and > 100\n",
    "  - ASAP: tags measured from the rat isotype controls <=75 \n",
    " - DOGMA: totalCTRLadt < 10 %(I think there is no such thing in the DOGMA-seq data, in the original code they do a regex search for \"Ctrl\" on the protein names, and there are none) - This was an error: index was unnamed! There are isotype controls!\n",
    " - ~((CD8_counts > 30) & (CD4_counts > 100))\n",
    " - don't remove high counts, as these are most likely the cells of interest..\n",
    " \n",
    "\n",
    "#### RNA (only applicable to DOGMA-seq datasets)\n",
    " - pct_mito < 30\n",
    " - n_total_rna > 1000 ~< (10^4.5)~\n",
    " - Use data adaptive threshold as upper threshold instead Q3 + 1.5x IQR (after having removed/excluding cells with low counts (<= 1000)!) n_total_rna <= q3 + 1.5 * iqr for top boundary!\n",
    " - n_feature_rna > 500\n",
    " - drop mitochondrial genes from analysis\n",
    " \n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "seeing-sandwich",
   "metadata": {},
   "source": [
    "## In practice\n",
    "### Downloaded data to data/original\n",
    "See README.md in data/original/Mimitou2021\n",
    "\n",
    "### ATAC-specific (DOGMA-seq and ASAP-seq)\n",
    "Informative features called with ScregSeg, for orientation/explanation see [McGarvey et al., 2022](https://doi.org/10.1016/j.xgen.2021.100083) and [tutorials](https://github.com/BIMSBbioinfo/scregseg/). In a nutshell, ScregSeg is a computational tool based on Hidden Markov Models that can be used as an alternative to peak calling for finding informative features for ATAC-seq data.\n",
    "\n",
    "#### ATAC prior to notebooks\n",
    "Provided fragment files for both DOGMA-seq and ASAP-seq data are gzipped BED files, however, ArchR, which is used for initial filtering requires 10x-like fragment files. These are sorted BED files that are bgzipped!\n",
    "\n",
    "To this end, I need to unzip, sort, and bgzip all fragment files. These are in /data/derived!\n",
    "\n",
    "```01_TCU_fragments_to_10x_fragments.sh```\n",
    "\n",
    "#### Filtering of ATAC data with ArchR\n",
    "```02_TCU_ATAC_filtering_ArchR.R```\n",
    "\n",
    "Initial filtering based on number of fragments and TSS enrichment.\n",
    "\n",
    "\n",
    "#### Scregseg first pass (shell script)\n",
    "For details see: ```03_TCU_ScregSeg_first_pass.sh```\n",
    "\n",
    "Settings for individual steps:\n",
    "- scregseg make_tile\n",
    "    - 1000 kb bins\n",
    "    - only autosomes \n",
    "- scregseg fragments_to_counts\n",
    "- scregseg subset\n",
    "    - subsets to only those cells meeting criterion of ```02_TCU_ATAC_filtering_ArchR.R``` (> 1000 fragments and TSS enrichment > 4)  \n",
    "- scregseg filter\n",
    "    - only keep regions with at least 1 count (after subsetting) across all cells and binarize data\n",
    "\n",
    "#### 04_ATAC_concatenate_scregseg_results.ipynb\n",
    "```04_TCU_ATAC_concatenate_scregseg_results.ipynb```\n",
    "\n",
    "#### Scregseg second pass (shell script)\n",
    "```05_TCU_ScregSeg_second_pass_50_states.sh```\n",
    "- scregseg fit_segment\n",
    "    - 7 random runs\n",
    "    - HMM with 50 states\n",
    "    - 300 iteration starting from random initial parameters for each run\n",
    "    \n",
    "#### 06_TCU_Explore_scregseg_second_pass.ipynb\n",
    "```06_TCU_Explore_scregseg_second_pass.ipynb```\n",
    "Determine thresholds for exporting informative regions\n",
    "\n",
    "#### Scregseg third pass (shell script)\n",
    "```07_TCU_ScregSeg_third_pass_50_states.sh```\n",
    "- scregseg seg_to_bed\n",
    "    - use regions of states that cover at most 1.5% of the genome as informative regions/features and that reach a posterior decoding probability of at least 0.9\n",
    "\n",
    "### GEX- and ADT-specific steps\n",
    "Detailed in ```08_Preprocess_GEX_and_ADT_data_and_combine_with_ATAC.ipynb```\n",
    "\n",
    "Combination of data."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "lpr_scregseg",
   "language": "python",
   "name": "lpr_scregseg"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
